src/components/HexagonGrid.tsx

```diff
@@ -81,11 +86,37 @@ function HexagonGrid(props: HexagonGridProps) {
     return dimensions;
   };
 
-  function testOnClick(hexagon: HexData) {
-    console.log("clicked! now", hexagon);
+  function updateHexagon(hexagon: HexData) {
+    const hex = hexagons.find(hex => hex.index === hexagon.index)
+
+    if (!hex) {
+      const newHexagon: HexData = {
+        index: hexagon.index,
+        terrain: brushSelection.terrain ? brushSelection.terrain : Terrain.NORMAL,
```

Comment:
can also write `brushSelection.terrain || Terrain.NORMAL`

However, I think the better solution is to leave it blank if not set. Just set it

`terrain: brushSelection.terrain`

Reply:
I get thrown an error if I just set it as
`terrain: brushSelection.terrain` 
cannot be undefined. Is it preferred to set it as 
`terrain: brushSelection.terrain || Terrain.NORMAL` or change Hexdata so terrain is optional.
`

Reply:
changing Hexdata so terrain is optional sounds right.